sync/course-db.js

```diff
@@ -1208,6 +1208,32 @@ async function validateAssessment(assessment, questions) {
     });
   });
 
+  // Ensure at least one role can assign roles before and during an assessment
+  if (assessment.groupRoles) {
```

Comment:
Is it worth doing some additional validation here? For instance, asserting that `role.minimum < groupMaxSize` and `role.maximum < groupMaxSize`?